merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/remotes/postgres/GetSimulatedActivitiesAction.java

```diff
@@ -69,14 +70,20 @@ public GetSimulatedActivitiesAction(final Connection connection) throws SQLExcep
     }
 
     // Since child IDs are not stored, we assign them by examining the parent ID of each activity
-    activities.forEach((id, activity) -> {
-      if (activity.parentId() == null) return;
-      activities.get(activity.parentId()).childIds().add(id);
-    });
+    activities.forEach(
+        (id, activity) -> activity
+            .parentId()
+            .ifPresent(parentId -> activities.get(parentId).childIds().add(id)));
 
     return activities;
   }
 
+  private Optional<Long> readOptionalLong(final ResultSet resultSet, final int index) throws SQLException {
+    final var value = resultSet.getLong(index);
+    if (resultSet.wasNull()) return Optional.empty();
+    return Optional.of(value);
```

Comment:
Following @camargo's comment this maybe should be `Optional.ofNullable()`? I'd hope `wasNull()` above this line would be enough but maybe there's some weird behaviour going on here? Sorry for the cursory/shallow observation here.

Reply:
Yeah, `resultSet.getLong()` returns 0 even if the value was null, so it can't be null there.

Reply:
Turns out the ResponseSerializer had to be updated. It was correctly using `serializeNullable()`, but it was calling `.id()` on the potentially null `ActivityInstanceId` value, so the `.id()` call had to be moved to the first argument to `serializeNullable()`.